### ISSUE TITLE
Can't run unit test after upgrade

### ISSUE THREAD

**Issue by Osama-Mater** [InsertKoinIO/koin#1499]
**Describe the bug**
After I upgraded the koin test deps to the latest stable version I can't run the unit-test again.

kotlin 1.8.0
gradle 7.3.1

```
koinTest      : "io.insert-koin:koin-test:3.3.2",
koinTestJunit4: "io.insert-koin:koin-test-junit4:3.3.2",
koinTestJunit5: "io.insert-koin:koin-test-junit5:3.3.2",
```

> FAILURE: Build failed with an exception.
> * What went wrong:
> Execution failed for task ':app:kaptGenerateStubsDevDebugUnitTestKotlin'.
> > Could not resolve all files for configuration ':app:devDebugUnitTestCompileClasspath'.
>    > Could not resolve org.jetbrains.kotlin:kotlin-test-junit5:1.8.0.
>      Required by:
>          project :app
>          project :app > org.jetbrains.kotlin:kotlin-test:1.8.0
>       > Module 'org.jetbrains.kotlin:kotlin-test-junit5' has been rejected:
>            Cannot select module with conflict on capability 'org.jetbrains.kotlin:kotlin-test-framework-impl:1.8.0' also provided by [org.jetbrains.kotlin:kotlin-test-junit:1.7.21(junitApi)]
>    > Could not resolve org.jetbrains.kotlin:kotlin-test-junit:1.7.21.
>      Required by:
>          project :app > io.insert-koin:koin-test-jvm:3.3.2
>       > Module 'org.jetbrains.kotlin:kotlin-test-junit' has been rejected:
>            Cannot select module with conflict on capability 'org.jetbrains.kotlin:kotlin-test-framework-impl:1.7.21' also provided by [org.jetbrains.kotlin:kotlin-test-junit5:1.8.0(junit5Api)]
>

**Comment by arnaudgiuliani** [comment]
can you use the latest Kotlin 1.7.x instead? Koin 3.3.x is running for Kotlin 1.7.x

**Comment by Osama-Mattar** [comment]
@arnaudgiuliani  We are targeting the latest Kotlin version 1.8.0, also I tried to downgrade the Kotlin version to 1.7.21 and I was receiving the same error.

**Comment by arnaudgiuliani** [comment]
the is something in your classpath that causing a conflict, apparently? 🤔 Can you reactive dependencies, one by one?

**Comment by stantronic** [comment]
Had this same problem - was referenced here: https://github.com/InsertKoinIO/koin/issues/1526

which provides a workaround :)

**Comment by Osama-Mattar** [comment]
@arnaudgiuliani The issue still happening, and we are facing more issues with the latest Koin version for creating the instances with High end devices (eg. Galaxy S20, Galaxy S21, Galaxy Z fold. Galaxy Z flip, Pixel 6 Pro, Pixel 6) and Android 13 & 12.
Unfortunately we are considering replacing it with Hilt.

**Comment by rmcmk** [comment]
Looks like the `build.gradle` for `koin-test` uses the `implementation` scope instead of `testImplementation` which would include test dependencies in the build

**Comment by arnaudgiuliani** [comment]
Fixed in PR - deployed for 3.5.4
